### PR TITLE
feat: encapsulate `Typography` styles.

### LINKED ISSUE #382: Q: typography styles
As I was [playing around with ui-kit being used with e.g. Gatsby/Next.js](https://github.com/commercetools/ui-kit/issues/353#issuecomment-449629247), I noticed that the `Text` components use kind of "global styles", even though the styles are defined in the css file.

https://github.com/commercetools/ui-kit/blob/d1acba93d97419f8a85db51471edf602087515fa/src/components/typography/text/text.mod.css#L15-L70

I'm wondering about a couple of things:
- should we really target html tags instead of using specific classes?
- if we do want to have it like this, we should maybe move those selectors to our e.g. `reset.mod.css`

Thoughts?

### PR DESCRIPTION
**Pull request by montezume**
Closes #382

### CODE REVIEW

File: src/components/typography/typography.story.js [hunk 1]
```diff
@@ -66,15 +66,15 @@ storiesOf('Basics|Typography/Text', module)
...
-          'none',
-          'primary',
-          'information',
-          'secondary',
-          'positive',
-          'negative',
-          'inverted',
-        ])}
+        tone={select('Text tone', {
+          none: null,
```
Comment (montezume) [inline src/components/typography/typography.story.js]: fixes storybook proptype warning

File: materials/internals/reset.mod.css [hunk 2]
```diff
@@ -39,8 +30,3 @@ a {
   text-decoration: none;
   transition: color 0.2s ease-in;
 }
-
-p {
```
Comment (montezume) [inline materials/internals/reset.mod.css]: making reset.mod.css a bit lighter 👍

File: rollup.config.js [hunk 3]
```diff
@@ -49,6 +50,7 @@ const postcssPlugins = [
     preserve: false,
     importFrom: { 'custom-properties': customProperties },
   }),
+  postcssNesting(),
```
Comment (tdeekens) [inline rollup.config.js]: Nice. This removes the need to `compose` even more.

File: src/components/typography/typography.story.js (hunk 1 above)
Comment (dferber90) [reply src/components/typography/typography.story.js]: Could it also be fixed by providing a default value to `select`?

File: src/components/typography/text/text.mod.css [hunk 4]
```diff
@@ -12,97 +12,110 @@
...
 }
 
-h4 {
+h4.subheadline {
   font-size: var(--font-h4);
   font-weight: normal;
 }
 
-h5 {
+h5.subheadline {
```
Comment (dferber90) [inline src/components/typography/text/text.mod.css]: Could we remove those `h1` prefixes from the selectors? Our JS code should ensure that they're applied to the right elements. I wouldn't consider it part of the CSS's job.

**Comment by montezume** [comment]
Yay for Percy 🎉

File: src/components/typography/text/text.mod.css [hunk 5]
```diff
@@ -118,3 +131,43 @@ small {
...
+    color: var(--color-gray-60);
+  }
+
+  &.positive {
+    color: var(--color-green-25);
+  }
+
+  &.negative {
+    color: var(--color-red);
+  }
```
Comment (dferber90) [inline src/components/typography/text/text.mod.css]: Could we define separate top-level classes instead? Then we might not need the `postcss-nesting` plugin as well.

So something like

```css
.bold {
  font-weight bold;
}

.positive {
   color: var(--color-green-25);
 }

.subheadline {
   margin: 0;
   color: var(--font-color);
 }
```

We are using the same "modifier" values from what I can tell, so this should reduce the amount of CSS as well.

Comment (montezume) [reply src/components/typography/text/text.mod.css]: done, I removed `postcss-nesting` and updated the PR to simplify the CSS. VRT still passing. I love VRT.

File: src/components/typography/text/text.mod.css (hunk 4 above)
Comment (montezume) [reply src/components/typography/text/text.mod.css]: We still need it for the font size though right? either that or we need two different classnames.

Comment (dferber90) [reply src/components/typography/text/text.mod.css]: Ah yeah, I missed that part. I'm okay with either approach then 👍

**Pull request merged**